### DISCUSSION TITLE
Warning: data for page "/" (path "/") is 736 KB which exceeds the threshold of 128 kB, this amount of data can reduce performance. See more info here: https://nextjs.org/docs/messages/large-page-data

### DISCUSSION THREAD

**Discussion by kulsumansari** [vercel/next.js#40606] [Help]
I am using next 12.0.9 version. There is a need of large amount of data on page.
What could be possible action I can take to resolve this warning. Also is there a way to disable/ increase threshold data size.

**Comment by icyJoseph** [comment]
> What could be possible action I can take to resolve this warning. 

The way to tackle this is to reduce the size of the props passed to your component from your data fetching function. That is to say `getStaticProps` passing data to a page component.

Perhaps not all parts of the data passed to the component are needed for the first render? 

Some of it might be fetched client side, even pre-loaded so that is available ASAP, see https://swr.vercel.app/docs/prefetching#top-level-page-data for example.

If all the data, 100%, is absolutely needed, then we are out of luck. 

> Also is there a way to disable/ increase threshold data size.

It is an estimate/heuristic that sits at 128 kB, so you are kind of way over that, and because of that perhaps silencing it is not a good idea.

That being said, I do believe you can add an experimental flag to your `next.config.js`. Keep in mind that experimental stuff is subject to change:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  experimental: {
    largePageDataBytes: 800 * 1000,
  },
};

module.exports = nextConfig;
```

The above sets the default at 800kB, which should be enough for your case, but I'd honestly let that warning be and acknowledge it.
